test/controllers/EvidenceDownloadController.test.ts

```diff
@@ -1,167 +1,289 @@
+        appData: Partial<ApplicationData>,
+        configureApp: (fileTransferService: FileTransferService, appealsService: AppealsService) => Application
     };
 
-    const readable = new Readable();
-    readable.push('');
-    readable.push(null);
+    const appConfigs: AppConfig[] = [
+        {
+            testDescription: 'As an Internal user',
+            appData: { appeal: internalUserAppeal },
+            configureApp: createInternalUserAppConfig
+        },
+        {
+            testDescription: 'As as External user',
+            appData: { appeal: externalUserAppeal },
+            configureApp: createExternalUserAppConfig
+        }
+    ];
 
-    const fileTransferServiceProxy = (downloadResult: Promise<Readable>,
-                                      metadata: FileMetadata): FileTransferService => {
+    function generateTests(appConfig: AppConfig): void {
 
-        const proxy = Substitute.for<FileTransferService>();
-        // @ts-ignore
-        proxy.download(Arg.any()).mimicks(async (fileId: string) => downloadResult);
-        // @ts-ignore
-        proxy.getFileMetadata(Arg.any()).mimicks(async (fileId: string) => Promise.resolve(metadata));
-        return proxy;
-    };
+        const { testDescription, appData, configureApp } = appConfig;
 
-    it('should render the prompt page correctly', async () => {
+        describe(`${testDescription}`, () => {
 
-        const fileTransferService = Substitute.for<FileTransferService>();
+            const FILE_ID = DEFAULT_ATTACHMENTS[0].id;
+            const APPEAL_ID = '345';
+            const DOWNLOAD_PROMPT_URL = `${DOWNLOAD_FILE_PAGE_URI}/prompt/${FILE_ID}?a=${APPEAL_ID}&c=NI000000`;
+            const EXPECTED_DOWNLOAD_LINK_URL = `${DOWNLOAD_FILE_PAGE_URI}/data/${FILE_ID}/download?a=${APPEAL_ID}&c=NI000000`;
 
-        await request(createDefaultApp(fileTransferService))
-            .get(DOWNLOAD_PROMPT_URL)
-            .then(res => {
-                expect(res.status).to.eq(200);
-                expect(res.text).to.include(`href="${EXPECTED_DOWNLOAD_LINK_URL}"`);
-            });
+            const contentDisposition = `attachment; filename=${DEFAULT_ATTACHMENTS[0].name}`;
 
-    });
+            const expectedDownloadErrorHeading = 'The file can not be downloaded at this moment';
+            const expectedDownloadErrorMessage = 'Please try again later';
 
-    it('should start downloading the file when the file is valid', async () => {
+            const metadataClean: FileMetadata = {
+                av_status: 'clean',
+                content_type: DEFAULT_ATTACHMENTS[0].contentType,
+                id: DEFAULT_ATTACHMENTS[0].id,
+                name: DEFAULT_ATTACHMENTS[0].name,
+                size: DEFAULT_ATTACHMENTS[0].size
+            };
 
-        const fakeFileTransferProxy =
-            fileTransferServiceProxy(Promise.resolve(readable), metadataClean);
+            it('should render the prompt page correctly', async () => {
```

Comment:
As there are two GET endpoints in this controller, I feel we should split the group of tests into two